src/entry-client.js

```diff
@@ -179,26 +167,10 @@ export default function initializeClient(createApp, clientOpts) {
                 router.getMatchedComponents(to)
                     .filter(c => 'vuex' in c)
                     .filter(c => shouldProcessRouteUpdate(c, fetchDataArgs, spruDefaults))
-                    .forEach((c) => {
-                        const name = getModuleName(c, to);
-                        const existingModule = find(registeredModules, { name });
-                        if (existingModule) {
-                            // We already have this module registered, update the
-                            // index to mark it as recent
-                            opts.logger.info('Skipping duplicate Vuex module registration:', name);
-                            existingModule.index = moduleIndex++;
-                        } else {
-                            opts.logger.info('Registering dynamic Vuex module:', name);
-                            // This module may have been registered outside of the
-                            // routing flow, so only register it with Vuex if needed -
-                            // but add it to our tracking of registeredModules regardless
-                            if (get(store, `_modulesNamespaceMap.${name}/`) == null) {
-                                store.registerModule(name, c.vuex.module, {
-                                    preserveState: store.state[name] != null,
-                                });
-                            }
-                            registeredModules.push({ name, index: moduleIndex++ });
```

Comment:
No longer doing any of this module tracking on registration, we will simply attempt to remove from the `from` components one route later via `queuedRemovalModules`